client/src/game/reducers/cardDeductionReducer.ts

```diff
@@ -0,0 +1,131 @@
+import * as deckRules from '../rules/deck';
+import { GameAction } from '../types/actions';
+import CardState from '../types/CardState';
+import GameMetadata from '../types/GameMetadata';
+import Variant from '../types/Variant';
+
+const cardDeductionReducer = (
```

Comment:
The parameter order is wrong for this to be a reducer.
newThing = reduce(oldThing, action, other params)
so
newDeck = cardDeductionReducer(oldDeck, action, hands, metadata)
Note metadata already includes the variant name, no need to pass again.